File: manual/code/generate_strategies/IR(summary)/java/java/function_pandas_standardize.java
package com.example;

import org.apache.commons.math3.stat.descriptive.moment.Mean;
import org.apache.commons.math3.stat.descriptive.moment.StandardDeviation;
import tech.tablesaw.api.DoubleColumn;
import tech.tablesaw.api.Table;

public class StandardizeData {
    public String standardize(Double valueA, Double valueB) {
        if (valueA == null || valueB == null) {
            return "A: None, B: None";
        }

        
        Table table = Table.create("data");
        table.addColumns(
                DoubleColumn.create("A", new Double[]{valueA}),
                DoubleColumn.create("B", new Double[]{valueB})
        );

        
        Mean mean = new Mean();
        StandardDeviation stdDev = new StandardDeviation();

        double meanA = mean.evaluate(new double[]{valueA});
        double stdA = stdDev.evaluate(new double[]{valueA});
        double meanB = mean.evaluate(new double[]{valueB});
        double stdB = stdDev.evaluate(new double[]{valueB});

        
        double standardizedA = (stdA != 0) ? (valueA - meanA) / stdA : 0;
        double standardizedB = (stdB != 0) ? (valueB - meanB) / stdB : 0;

        
        standardizedA = Double.isNaN(standardizedA) ? 0 : standardizedA;
        standardizedB = Double.isNaN(standardizedB) ? 0 : standardizedB;

        return String.format("A: %f, B: %f", standardizedA, standardizedB);
    }
}

File: manual/code/test_suites/cpp/FunctionBuildTest/src/function_pandas_standardize.cpp
#include <cmath>
#include <string>
#include <vector>
#include <algorithm>

class StandardizeData {
public:
    std::string standardize(double valueA, double valueB) {
        if (std::isnan(valueA) || std::isnan(valueB)) {
            return "A: None, B: None";
        }

        // 计算均值和标准差
        double meanA = valueA;
        double stdA = 0.0;
        double meanB = valueB;
        double stdB = 0.0;

        // 标准化计算
        double standardizedA = (stdA != 0) ? (valueA - meanA) / stdA : 0;
        double standardizedB = (stdB != 0) ? (valueB - meanB) / stdB : 0;

        // 处理可能的NaN情况
        standardizedA = std::isnan(standardizedA) ? 0 : standardizedA;
        standardizedB = std::isnan(standardizedB) ? 0 : standardizedB;

        char buffer[100];
        snprintf(buffer, sizeof(buffer), "A: %f, B: %f", standardizedA, standardizedB);
        return std::string(buffer);
    }
};